extensions/wikia/Helios/User.class.php

```diff
@@ -190,13 +192,14 @@ public static function authenticate( $username, $password ) {
 			'result' => $result,
 			'exception' => $throwException,
 			self::AUTH_METHOD_NAME => $authMethod,
+			self::STATUS_NAME => $status,
 		];
 
 		if ( $throwException ) {
 			throw $throwException;
 		}
 
-		if ( !empty( $loginInfo ) && $authMethod != self::AUTH_TYPE_RESET_PW ) {
+		if ( isset( $loginInfo->access_token ) && $authMethod != self::AUTH_TYPE_RESET_PW ) {
```

Comment:
This seemed like a bug since `loginInfo` contains error information when helios returns non-200, but doesn't have the `access_token` field.